share/transformers/org_socialscienceregistry.py

```diff
@@ -0,0 +1,226 @@
+from collections import OrderedDict
+
+import json
+
+from share.transform.chain import ctx, ChainTransformer
+from share.transform.chain import links as tools
+from share.transform.chain.parsers import Parser
+
+
+class AgentIdentifier(Parser):
+    uri = tools.IRI(ctx)
+
+
+class WorkIdentifier(Parser):
+    uri = tools.IRI(ctx)
+
+
+class Organization(Parser):
+    name = ctx
+
+
+class Publisher(Parser):
+    agent = tools.Delegate(Organization, ctx)
+
+
+class Institution(Parser):
+    name = ctx
+
+
+class IsAffiliatedWith(Parser):
+    related = tools.Delegate(Institution)
+
+
+class Person(Parser):
+    given_name = tools.ParseName(tools.Try(ctx.name)).first
+    family_name = tools.ParseName(tools.Try(ctx.name)).last
+    identifiers = tools.Map(tools.Delegate(AgentIdentifier), tools.Try(ctx.email))
+
+
+class Creator(Parser):
+    agent = tools.Delegate(Person, ctx)
+
+
+class Subject(Parser):
+    name = ctx
+
+
+class Tag(Parser):
+    name = ctx
+
+
+class ThroughTags(Parser):
+    tag = tools.Delegate(Tag, ctx)
+
+
+class ThroughSubjects(Parser):
+    subject = tools.Delegate(Subject, ctx)
+
+
+def process_keywords(text):
+    text = json.loads(text)
+    text = [item for item in text if item]
+    return text
+
+
+class Registration(Parser):
+    title = tools.Try(ctx['general-information']['title'])
+    description = tools.Try(ctx['additional-trial-info']['abstract'])
+    date_updated = tools.ParseDate(tools.Try(ctx['general-information']['last-updated']))
+    date_published = tools.ParseDate(tools.Try(ctx['general-information']['published-at']))
+    related_agents = tools.Map(tools.Delegate(Creator), tools.Try(ctx.pi))
+    identifiers = tools.Map(
+        tools.Delegate(WorkIdentifier),
+        tools.Concat(
+            tools.Try(ctx['general-information']['url'])
```

Comment:
The IRI parser should probably be moved down here so that a through identifier isn't created if the given identifier isn't valid.